src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -166,18 +167,79 @@ public void EmitDebugLoc(int nativeOffset, int fileId, int linueNumber, int colN
         }
 
         [DllImport(NativeObjectWriterFileName)]
-        private static extern void FlushDebugLocs(IntPtr objWriter, string methodName, int methodSize);
-        public void FlushDebugLocs(int methodSize)
+        private static extern void EmitDebugVar(IntPtr objWriter, UInt32 varNumber, string name, UInt32 typeIndex, bool isParam, byte[] ranges, Int32 rangeCount);
+
+        public void LoadValueToArray(Int32 value, byte[] array, ref int pos)
+        {
+            for (int i = 0; i < sizeof(Int32); i++)
+            {
+                array[pos] = (byte) (value & 0xff);
+                value = value >> 8;
+                pos++;
+            }
+        }
+
+        public void LoadValueToArray(Int64 value, byte[] array, ref int pos)
+        {
+            for (int i = 0; i < sizeof(Int64); i++)
+            {
+                array[pos] = (byte)(value & 0xff);
+                value = value >> 8;
+                pos++;
+            }
+        }
+
+        public void EmitDebugVar(DebugVarInfo debugVar)
+        {
+            int rangeCount = debugVar.ranges.Count;
+            int rangeSize;
+
+            unsafe { rangeSize = sizeof(NativeVarInfo); };
+
+            byte[] hackedRanges = new byte[rangeCount * rangeSize];
```

Comment:
Can you add a comment why this is not simply marhalled as an array of NativeVarInfo?